src/rest.cpp

```diff
@@ -454,12 +484,39 @@ static bool rest_block(const std::any& context,
 
 static bool rest_block_extended(const std::any& context, HTTPRequest* req, const std::string& uri_part)
 {
-    return rest_block(context, req, uri_part, TxVerbosity::SHOW_DETAILS_AND_PREVOUT);
+    return rest_block(context, req, uri_part, TxVerbosity::SHOW_DETAILS_AND_PREVOUT, /*part_offset=*/0, /*part_size=*/std::nullopt);
 }
 
 static bool rest_block_notxdetails(const std::any& context, HTTPRequest* req, const std::string& uri_part)
 {
-    return rest_block(context, req, uri_part, TxVerbosity::SHOW_TXID);
+    return rest_block(context, req, uri_part, TxVerbosity::SHOW_TXID, /*part_offset=*/0, /*part_size=*/std::nullopt);
+}
+
+static bool rest_block_part(const std::any& context, HTTPRequest* req, const std::string& uri_part)
+{
+    size_t part_offset{0};
+    std::optional<size_t> part_size{};
+    try {
+        const auto part_offset_str = req->GetQueryParameter("offset");
+        if (part_offset_str.has_value()) {
+            auto opt = ToIntegral<size_t>(*part_offset_str);
+            if (opt.has_value()) {
+                part_offset = opt.value();
+            } else {
+                return RESTERR(req, HTTP_BAD_REQUEST, strprintf("Block part offset is invalid: %s", *part_offset_str));
+            }
+        }
+        const auto part_size_str = req->GetQueryParameter("size");
+        if (part_size_str.has_value()) {
+            part_size = ToIntegral<size_t>(*part_size_str);
+            if (!part_size.has_value()) {
+                return RESTERR(req, HTTP_BAD_REQUEST, strprintf("Block part size is invalid: %s", *part_size_str));
+            }
```

Comment:
The idiomatic way to use the `std::optional` is to guard in the if condition for its existence.
`.value()` only really makes sense over `*` when we need it to throw - which it can't if we're already guarding.
Also, the first condition has an if-exists-assign-else-return format, while the second checks for non-existence and assigns to external.
We could unify and simplify this to something like:
```suggestion
        if (const auto part_offset_str{req->GetQueryParameter("offset")}) {
            if (const auto opt{ToIntegral<size_t>(*part_offset_str)}) {
                part_offset = *opt;
            } else {
                return RESTERR(req, HTTP_BAD_REQUEST, strprintf("Block part offset is invalid: %s", *part_offset_str));
            }
        }
        if (const auto part_size_str{req->GetQueryParameter("size")}) {
            if (const auto opt{ToIntegral<size_t>(*part_size_str)}) {
                part_size = opt;
            } else {
                return RESTERR(req, HTTP_BAD_REQUEST, strprintf("Block part size is invalid: %s", *part_size_str));
            }
        }
```